incentives.py

```diff
@@ -0,0 +1,311 @@
+from pgmpy.models import BayesianModel
+import networkx as nx
+import copy
+from itertools import compress
+from typing import List
+
+
+##
+class Information:
+    """Single agent Information incentive Class
+    Criterion for Information incentive on X:
+    (i) X is d-connected to U | Fa_D\{X}
+    (ii) U∈Desc(D) (U must be a descendent of D)
+    """
+
+    def __init__(self, model, decision_nodes, utility_nodes):
+    # pass in model, decision nodes and utility nodes of agent being considered    
+        try:
+            assert isinstance(model, BayesianModel)
+        except AssertionError:
+            raise AssertionError("Observation instantiation's first argument must be a BayesianModel")
+
+        self.dag = model
+        self.graph = self.dag.to_directed()
+        self.decision_nodes = decision_nodes
+        self.utility_nodes = utility_nodes
+        self.all_nodes = self.dag.nodes
+
+        try:
+            assert set(self.decision_nodes).issubset(set(self.all_nodes))
+            assert set(self.utility_nodes).issubset(set(self.all_nodes))
+        except AssertionError:
+            raise AssertionError("Some or all of the Decision and Utility nodes are not present on the graph")
+
+    # def __repr__(self):  #maybe create this function
+    #     return
+
+    def inf_node(self, node):
+        # returns True if a node faces an observation incentive
+        try:
+            assert self.utility_nodes[0] in nx.descendants(self.dag, self.decision_nodes[0])
+        except AssertionError:
+            raise AssertionError(
+                f"utility node {self.utility_nodes[0]} is not a descendent of decision node {self.decision_nodes[0]}")
+
+        if node not in self.utility_nodes + self.decision_nodes:
+            con_nodes = self.decision_nodes + self.dag.get_parents(self.decision_nodes[0])  # nodes to be conditioned on
+            if node in con_nodes:  # remove node from condition nodes
+                con_nodes.remove(node)
+            if self.dag.is_active_trail(node, self.utility_nodes[0], con_nodes):
+                return True
+            else:
+                return False
+
+    def all_inf_inc_nodes(self):
+        # returns all nodes in the DAG that the the decision maker has an observation incentive for.
+        inf_list = []
+        for x in list(self.dag.nodes):
+            if self.inf_node(x):
+                inf_list.append(x)
+
+        return f"The nodes which are facing an information incentive from this agent are {inf_list}"
+
+
+##
+
+class Response:
```

Comment:
To me, it's a bit weird that you're making an object that takes in and contains a model, then returns all the incentives in the graph. 

Suppose we made this a method of the cid class. Then couldn't we do away with the whole __init__ bit?

Reply:
Yep - I did as you said. incentives.py has been deleted